shell/src/App.js

```diff
@@ -28,9 +48,36 @@ const App = () => (
                 </Layer>
             }
         >
-            <D2App config={appConfig} />
+            <D2App config={config} />
         </React.Suspense>
     </AppAdapter>
 )
 
-export default App
+App.propTypes = {
+    config: PropTypes.object,
+}
+
+const Plugin = ({ config }) => {
+    return (
+        <PluginOuterErrorBoundary>
```

Comment:
this exists, so that an error in PluginLoader could be caught. I'm not really sure that this is ultimately necessary* as most of the logic there is just to handle the post-robot communication which is set to silently fail, so as not to impede plugin's usability if communication is not needed. If we do want an error boundary here, I guess we could reuse the same one from app-platform/adapter?

* right now, it's necessary based on how required props are being handled, but I think this should probably change